examples/md-flexible/src/configuration/YamlParser.h

```diff
@@ -15,11 +17,153 @@
- * @note FIXME: at the moment false is never returned and the parser just ungracefully crashes.
  */
 bool parseYamlFile(MDFlexConfig &config);
+
+/**
+ * Parses an input where exactly one option is expected and throws an error if a sequence of multiple options is given
+ * @param node The current YAML::Node, that should be parsed
+ * @param errMsg The error message thrown if multiple options are passed
+ * @return String representation of the parsed node
+ */
+const std::string parseSequenceOneElementExpected(const YAML::Node node, const std::string &errMsg);
+
+/**
+ * Creates an error message for all non-object-keys
+ * @param mark The Yaml-Mark for printing line- and column number
+ * @param key The key that caused the error
+ * @param errorMsg Message thrown by an exception in yaml-cpp, in the parser, or in AutoPas
+ * @param expected The expected value of the key
+ * @param description The parameter description of the key
+ * @return String representation of the parsed node
+ */
+const std::string makeErrorMsg(const YAML::Mark &mark, const std::string &key, const std::string &errorMsg,
+                               const std::string &expected, const std::string &description);
+
+/**
+ * Parses the scalar value of a key in a Object-Node of a YAML-config.
+ * @param node root-YAML-node of an Object.
+ * @param key The key to parse.
+ * @param objectErrors Vector to store all errors during parsing of one object
+ * @return Parsed value of key. Throws a runtime_error if key could not be parsed.
+ */
+template <typename T>
+const T parseObjectValueSingle(const YAML::Node node, const std::string &key, std::vector<std::string> &objectErrors) {
+  T value;
+  try {
+    value = node[key].as<T>();
+  } catch (const std::exception &e) {
+    std::stringstream ss;
+    ss << "Error parsing " << key << ". Make sure that key \"" << key
+       << "\" exists and has the expected value: " << type2Str<T>();
+    objectErrors.push_back(ss.str());
+  }
+  return value;
+}
+
+/**
+ * Parses the sequence value of a key in a Object-Node of a YAML-config
+ * @param node root-YAML-node of an Object.
+ * @param key The key to parse.
+ * @param objectErrors Vector to store all errors during parsing of one object
+ * @return Parsed value of key. Throws a runtime_error if key could not be parsed.
+ */
+template <typename T, size_t S>
+const std::array<T, S> parseObjectValueSequence(const YAML::Node node, const std::string &key,
+                                                std::vector<std::string> &objectErrors) {
+  std::array<T, S> value;
+  try {
+    YAML::Node n = node[key];
+    for (int i = 0; i < S; i++) {
+      value[i] = n[i].as<T>();
+    }
+
+  } catch (const std::exception &e) {
+    std::stringstream ss;
+    ss << "Error parsing " << key << ". Make sure that key \"" << key << "\" exists and has the expected value: "
+       << "YAML-sequence of " << std::to_string(S) << " " << type2Str<T>() << " values.";
```

Comment:
```suggestion
       << "YAML-sequence of " << std::to_string(S) << " " << typeToStr<T>() << " values.";
```
Follow up from suggestion above.

Reply:
done [679ee4c](https://github.com/AutoPas/AutoPas/pull/732/commits/679ee4ce977ef550244be4f665be6cfc941b89ea)